flake.nix

```diff
@@ -0,0 +1,75 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+
+        # Read version from Cargo.toml
+        cargoToml = builtins.fromTOML (builtins.readFile ./crates/tattoy/Cargo.toml);
+        version = cargoToml.package.version;
+
+        nativeBuildInputs = with pkgs; [
+          rustc
+          cargo
+          pkg-config
+        ];
+        buildInputs = with pkgs; [
+          dbus
+          xorg.libxcb
+        ];
+
+        tattoy = pkgs.rustPlatform.buildRustPackage {
+          pname = "tattoy";
+          inherit version;
+
+          src = ./.;
+
+          cargoLock = {
+            lockFile = ./Cargo.lock;
+            allowBuiltinFetchGit = true;
+          };
+
+          checkInputs = with pkgs; [
+            bash
+            nano
+            procps
+          ];
+          checkPhase = ''
+            runHook preCheck
+            cargo build --all
+            cargo nextest run
+            runHook postCheck
+          '';
```

Comment:
I wonder if there's a way to encode this without adding a custom checkPhase. I tried

```
          cargoTestFlags = [ "--workspace" ];
          checkType = "debug";
          useNextest = true;
```

but this fails more of the e2e tests (12 instead of 10). I'm not really sure why.